src/ps-stats/statsService.ts

```diff
@@ -0,0 +1,53 @@
+import fs = require("fs");
+import { PokemonUsage, SmogonFormat } from "./models";
+
+type DbData = { [id: string] : any[]; }
+enum StatsType { Leads = "leads" }
+
+export class StatsService {
+	
+	private database: { [id: string] : DbData; } = {};
+	
+	public getLeads(format: SmogonFormat): PokemonUsage[] {
+		this.ensureDataIsLoaded(StatsType.Leads, format, (data) => {
+			return data.data.rows
+				.sort((a: any[], b: any[]) => (a[2] - b[2]) * -1) // reverse by usage percentage
+				.slice(0, 10)
+				.map((itemData: any[]) => {
+					return {
+						name: itemData[1], 
+						usagePercentage: itemData[2], 
+						usageRaw: itemData[3] 
+					} as PokemonUsage
+				});
+		});
+
+		const fmt = this.getFormatKey(format);
+		return this.database[StatsType.Leads][fmt];
+	}
+
+	private ensureDataIsLoaded(statsType: string, format: SmogonFormat, callback?: (data: any) => any): void {
```

Comment:
Might want to comment this file a bit better, took me a bit to understand what was going on with the callback, ect here.

Reply:
Sure, it's the quite complex indeed. I would not bother much with this part since we'll move to the Showdown Statistics package in some time. Anyways, I'm writing comments to help anyone else in the mean while.